src/TaskProviderDune.ml

```diff
@@ -0,0 +1,95 @@
+open Import
+
+let taskDefinition = { Task.type_ = "dune" }
+
+let source = "dune"
+
+let problemMatchers = [| "ocamlc" |]
+
+(* the ocamlc matcher is not able to parse ocaml compiler errors unless they
+   follow the short style. *)
+let env = Js.Dict.fromList [ ("OCAML_ERROR_STYLE", "short") ]
+
+let startsWith ~s ~prefix =
+  try
+    let prefixLen = String.length prefix in
+    let start = String.sub s 0 prefixLen in
+    String.equal start prefix
+  with _ -> false
+
+let removePrefix ~s ~prefix =
+  let prefixLen = String.length prefix in
+  let len = String.length s - prefixLen in
+  String.sub s prefixLen len
+
+let folderRelativePath folders file =
+  Array.fold_left
+    (fun acc (folder : Folder.t) ->
+      match acc with
+      | Some _ -> acc
+      | None -> (
+        match startsWith ~s:file ~prefix:folder.uri.fsPath with
+        | false -> acc
+        | true -> Some (folder, removePrefix ~s:file ~prefix:folder.uri.fsPath)
+        ))
+    None folders
+
+let commandLine () =
+  let open Promise.O in
+  Toolchain.ofSettings () >>| function
+  | None -> "dune build"
+  | Some pm ->
+    let resources = Toolchain.makeResources pm in
+    let cmd, args = Toolchain.getCommand resources "dune" [ "build" ] in
```

Comment:
There seems to be a lot of duplication of this "dune" string. I propose that we have 2 constants:

* One for the binary name `dune`.
* Another one for the task type `dune`.